### PR TITLE
fix: cadvisor can pickup oom events

### PR DESCRIPTION
**Pull request by michaellzc**
Follow up https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/804

### Checklist

- [x] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)

### Test plan

tested on a live gke cluster

deploy sourcegraph as usual

deploy a stress test pod
```sh
apiVersion: v1
kind: Pod
metadata:
  name: stress
  namespace: sourcegraph
spec:
  containers:
    - name: stress
      image: progrium/stress
      command:
        - sleep
        - infinity
      resources:
        requests:
          memory: 100Mi
        limits:
          memory: 128Mi
```

exec into the pod and run
```sh
stress --cpu 1 --vm-bytes 300M --timeout 10s
```

oom events show up as expected

CleanShot 2022-05-05 at 15 48 49

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #121** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/121" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/121?utm_source=stack-comment).

File: charts/sourcegraph/templates/cadvisor/cadvisor.DaemonSet.yaml [hunk 1]
```diff
@@ -82,6 +82,11 @@ spec:
         - name: disk
           mountPath: /dev/disk
           readOnly: true
+        {{- if .Values.cadvisor.containerSecurityContext.privileged }}
+        - name: kmsg
+          mountPath: /dev/kmsg
+          readOnly: true
+        {{- end }}
```
Comment (michaellzc) [inline charts/sourcegraph/templates/cadvisor/cadvisor.DaemonSet.yaml:89]: maybe I am overcomplicating things.

without `privileged` and we mount the `/dev/kmsg`, users would see the below logs from `cadvisor`.

```
k logs daemonset/cadvisor                                                     14:05:31
Found 2 pods, using pod/cadvisor-qkdlk
W0505 21:05:23.014872       1 machine_libipmctl.go:63] There are no NVM devices!
W0505 21:05:23.068489       1 manager.go:296] Could not configure a source for OOM detection, disabling OOM events: open /dev/kmsg: operation not permitted
``

If we don't mount `/dev/kmsg`, users will always see something like

```
 Could not configure a source for OOM detection, disabling OOM events: open /dev/kmsg: no such file or directory
```

either one of them is not ideal, so I figure we should just don't mount unnecessary stuff when it has no used (`privielged=false`)

File: charts/sourcegraph/templates/cadvisor/cadvisor.DaemonSet.yaml [hunk 2]
```diff
@@ -119,6 +124,11 @@ spec:
       - name: disk
         hostPath:
           path: /dev/disk
+      {{- if .Values.cadvisor.containerSecurityContext.privileged }}
+      - name: kmsg
+        hostPath:
+          path: /dev/kmsg
+      {{- end }}
```
Comment (michaellzc) [inline charts/sourcegraph/templates/cadvisor/cadvisor.DaemonSet.yaml:131]: there's no docker's `device` equivalent in `Kubernetes`, so we have to mount it as a regular volume.

File: charts/sourcegraph/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ Use `**BREAKING**:` to denote a breaking change
 
 All notable changes to Sourcegraph are documented in this file.
 
+### Changed
+
+- **IMPORTANT** `cadvisor` can collect out of memory events happening to containers and it can be used to discover underprovisoned resources. As a result, `cadvisor` now has to run in `privileged` mode. If you have your own monitoring infrastructure, you may choose to disable `cadvisor` or set `cadvisor.containerSecurityContext.privileged=false` in your override file. [#121](https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/121)
```
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/CHANGELOG.md:11]:
```suggestion
- **IMPORTANT** `cadvisor` now defaults to run in `privileged` mode. This allows `cadvisor` to collect out of memory events happening to containers which can be used to discover underprovisoned resources. If you have your own monitoring infrastructure, you may choose to disable `cadvisor` or set `cadvisor.containerSecurityContext.privileged=false` in your override file. [#121](https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/121)
```

**Review by caugustus-sourcegraph — APPROVED** [review]

**Pull request merged**